src/client/dfuse/dfuse_core.c

```diff
@@ -64,6 +64,55 @@ dfuse_fs_send(struct dfuse_request *request)
 	return rc;
 }
 
+/* Inode record hash table operations */
+
+static bool
+ir_key_cmp(struct d_hash_table *htable, d_list_t *rlink,
+	   const void *key, unsigned int ksize)
+{
+	const struct dfuse_inode_record		*ir;
+	const struct dfuse_inode_record_id	*ir_id = key;
+
+	ir = container_of(rlink, struct dfuse_inode_record, ir_htl);
+
+	/* First check if the both parts of the OID match */
+	if (ir->ir_id.irid_oid.lo != ir_id->irid_oid.lo)
+		return false;
+
+	if (ir->ir_id.irid_oid.hi != ir_id->irid_oid.hi)
+		return false;
+
+	/* Then check if it's the same container (dfs struct) */
+	if (ir->ir_id.irid_dfs == ir_id->irid_dfs) {
+		return true;
+	}
+
+	/* Now check the container name */
+	if (strncmp(ir->ir_id.irid_dfs->dffs_cont,
+		    ir_id->irid_dfs->dffs_cont,
```

Comment:
Could just do a uuid_compare() here if UUIDs were stored instead of strings.

Reply:
Good to know, I was wondering that although I managed to make it so this line is almost never reached, within a container the ie_dfs pointer matches so the function returns there, this is only used for lookup() of an existing container, which is an area which needs optimisation anyway as we currently call container_connect() on that code path before discovering it's a duplicate.